server/src/internalClusterTest/java/org/elasticsearch/snapshots/MetadataLoadingDuringSnapshotRestoreIT.java

```diff
@@ -1,217 +0,0 @@
-        assertThat(getSnapshotsResponse.getSnapshots("repository"), hasSize(1));
-        assertGlobalMetadataLoads("snap", 0);
-        assertIndexMetadataLoads("snap", "docs", 0);
-        assertIndexMetadataLoads("snap", "others", 0);
-
-        // Getting the status of a snapshot loads indices metadata but not global metadata
-        SnapshotsStatusResponse snapshotStatusResponse =
-            client().admin().cluster().prepareSnapshotStatus("repository").setSnapshots("snap").get();
-        assertThat(snapshotStatusResponse.getSnapshots(), hasSize(1));
-        assertGlobalMetadataLoads("snap", 0);
-        assertIndexMetadataLoads("snap", "docs", 1);
-        assertIndexMetadataLoads("snap", "others", 1);
-
-        assertAcked(client().admin().indices().prepareDelete("docs", "others"));
-
-        // Restoring a snapshot loads indices metadata but not the global state
-        RestoreSnapshotResponse restoreSnapshotResponse = client().admin().cluster().prepareRestoreSnapshot("repository", "snap")
-                                                                                    .setWaitForCompletion(true)
-                                                                                    .get();
-        assertThat(restoreSnapshotResponse.getRestoreInfo().failedShards(), equalTo(0));
-        assertGlobalMetadataLoads("snap", 0);
-        assertIndexMetadataLoads("snap", "docs", 2);
-        assertIndexMetadataLoads("snap", "others", 2);
-
-        assertAcked(client().admin().indices().prepareDelete("docs"));
-
-        // Restoring a snapshot with selective indices loads only required index metadata
-        restoreSnapshotResponse = client().admin().cluster().prepareRestoreSnapshot("repository", "snap")
-                                                            .setIndices("docs")
-                                                            .setWaitForCompletion(true)
-                                                            .get();
-        assertThat(restoreSnapshotResponse.getRestoreInfo().failedShards(), equalTo(0));
-        assertGlobalMetadataLoads("snap", 0);
-        assertIndexMetadataLoads("snap", "docs", 3);
-        assertIndexMetadataLoads("snap", "others", 2);
-
-        assertAcked(client().admin().indices().prepareDelete("docs", "others"));
-
-        // Restoring a snapshot including the global state loads it with the index metadata
-        restoreSnapshotResponse = client().admin().cluster().prepareRestoreSnapshot("repository", "snap")
-            .setIndices("docs", "oth*")
-            .setRestoreGlobalState(true)
-            .setWaitForCompletion(true)
-            .get();
-        assertThat(restoreSnapshotResponse.getRestoreInfo().failedShards(), equalTo(0));
-        assertGlobalMetadataLoads("snap", 1);
-        assertIndexMetadataLoads("snap", "docs", 4);
-        assertIndexMetadataLoads("snap", "others", 3);
```

Comment:
Maybe keep this test and just removing the mentioned lines? Or assume that during a restore global state is loaded.
So that we ensure no global state is loaded when creating a snapshot or using get snapshot api and snapshot status api?

Reply:
Yes, it was silly of me to remove the test completely, I restored it and modified only required parts